src/consumer/index.ts

```diff
@@ -69,97 +58,111 @@ export class ConsumerRouter {
 
   public async stop(): Promise<void> {
     if (this.consumer != null) {
-      /* istanbul ignore next */
-      this.consumer.close((e) => {
-        console.error(e);
-      });
-      this.consumer = null;
+      await this.consumer.disconnect();
+    }
+  }
+
+  private async processMessage(topic: string, content: Input): Promise<void> {
+    return Promise.all(
+      this.routes
+        .filter((route) => topic === route.topic && (route.eventName == null || route.eventName === content.code))
+        .map((route) => {
+          debug(Debug.TRACE, 'Message received on route', route);
+          return route.callback(content, emit);
+        })
+    ).then((results) => {
+      if (results.length === 0) {
+        debug(Debug.DEBUG, 'Committing without match');
+      } else {
+        debug(Debug.DEBUG, 'Committing with', results.length, 'matches');
+      }
+    });
+  }
+
+  private initQueues(topics: string[]): void {
+    for (const topic of topics) {
+      this.queues[topic] = {
+        status: 'alive',
+        promises: []
+      };
     }
   }
 
   public async start(): Promise<void> {
     const config = getConfig();
 
-    if (config.consumer?.groupId == null || config.consumer.groupId.trim() === '') {
+    if (config.consumer == null || config.consumer.groupId == null || config.consumer.groupId.trim() === '') {
       throw new Error('Missing configuration config.consumer.groupId for consumer');
     }
 
     const groupId = config.consumer.groupId;
     const kafkaHost = config.host;
-    const fetchSizeInMB = config.consumer.fetchSizeInMB ?? DEFAULT.fetchSizeInMB;
     const onlyTesting = config.onlyTesting ?? DEFAULT.onlyTesting;
 
-    if (config.consumer.autoCommit != null) {
-      this.autoCommit = config.consumer.autoCommit;
-    }
-
     if (this.routes.length === 0) {
       throw new Error('Missing routes, please add minimum 1 route');
     }
     if (onlyTesting) {
       return Promise.resolve();
     } else {
-      return new Promise((resolve, reject) => {
-        const MB = 1024 * 1024;
-        this.consumer = new ConsumerGroup(
-          {
-            groupId,
-            kafkaHost,
-            autoCommit: this.autoCommit,
-            encoding: 'utf8',
-            fetchMaxWaitMs: 100,
-            fetchMaxBytes: MB * fetchSizeInMB
-          },
-          this.routes
-            .map((route) => route.topic)
-            .filter((value, index, array) => array.indexOf(value) === index)
-        );
-
-        this.consumer.on('connect', () => {
-          console.log('Event server ready');
-          resolve();
-        });
-
-        this.consumer.on('message', (message: Message) => {
+      const kafka = new Kafka({
+        brokers: kafkaHost.split(','),
```

Comment:
what happen if we define this value as a `string` array to avoid this split

Reply:
by convention kafka broker should be a string comma sepatared